node/src/components/storage.rs

```diff
@@ -351,14 +359,28 @@ impl Storage {
         Ok(match req {
             StorageRequest::PutBlock { block, responder } => {
                 let mut txn = self.env.begin_rw_txn()?;
-                let outcome = txn.put_value(self.block_db, block.hash(), &block, true)?;
+                if !txn.put_value(
+                    self.block_body_db,
+                    block.header().body_hash(),
+                    block.body(),
+                    true,
+                )? {
+                    error!("Could not insert block body for block: {}", block);
+                    txn.abort();
+                    return Ok(responder.respond(false).ignore());
+                }
+                if !txn.put_value(self.block_header_db, block.hash(), block.header(), true)? {
+                    error!("Could not insert block header for block: {}", block);
+                    txn.abort();
+                    return Ok(responder.respond(false).ignore());
+                }
```

Comment:
We should try to remove the body if we failed to insert the header.

Reply:
Right but this is in the case of a failure or some error to store the the block

Reply:
Yes, I understand that. I just want to make sure that we're not leaving any leftover, unnecessary state.

Reply:
I checked. It does revert it.

Reply:
Aaaah okay, sorry my mistake